playground/backend/api.js

```diff
@@ -56,7 +69,82 @@ async function doFetch(url) {
       'Referer': 'https://amp.dev/playground',
     },
   });
+
+  if (!response.ok) {
+    throw new Error(`Request to ${fetchUrl} could not complete successfully.`);
+  }
+
+  const contentType = response.headers.get('content-type');
+  if (!contentType.includes('text/html')) {
+    throw new Error(`${fetchUrl} is no HTML document.`);
+  }
+
   return response.text();
 }
 
+const limits = new LRU({
+  maxAge: RATE_LIMIT_TIME_FRAME,
```

Comment:
Should this have a maximum size?

Reply:
Have been pondering over this: one could flood the API with requests to random hosts to make a ongoing rate-limit fall out. But added a fairly high one with 1000 now.